src/nix/develop.cc

```diff
@@ -353,9 +386,42 @@ struct Common : InstallableCommand, MixProfile
             }
         }
 
+        if (buildEnvironment.providesStructuredAttrs()) {
+            fixupStructuredAttrs("sh", buildEnvironment.getAttrsSH(), rewrites, buildEnvironment, tmpDir);
```

Comment:
Instead of `extToEnvVar`, just pass the environment variable.
This is easier to understand and discoverable with grep.

```suggestion
            fixupStructuredAttrs("sh", "NIX_ATTRS_SH_FILE", buildEnvironment.getAttrsSH(), rewrites, buildEnvironment, tmpDir);
```